src/components/PeopleFilters/PeopleFilters.tsx

```diff
@@ -0,0 +1,119 @@
+import classNames from 'classnames';
+import { useCallback, memo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { centuriesList } from '../../constants/centuriesList';
+import { getSearchWith } from '../../utils/searchHelper';
+import { SearchLink } from '../SearchLink/SearchLink';
+
+export const PeopleFilters: React.FC = memo(() => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries');
+  const sex = searchParams.get('sex') || '';
+
+  const onQueryChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => (
+      setSearchParams(getSearchWith(searchParams, {
+        query: event.target.value.trim() || null,
+      }))
+    ), [searchParams],
+  );
+
+  return (
+    <nav className="panel">
+      <p className="panel-heading">Filters</p>
+      <p className="panel-tabs" data-cy="SexFilter">
+        <SearchLink
+          params={{ sex: null }}
+          className={classNames({ 'is-active': sex === '' })}
+        >
+          All
+        </SearchLink>
+        <SearchLink
+          params={{ sex: 'm' }}
+          className={classNames({ 'is-active': sex === 'm' })}
+        >
+          Male
+        </SearchLink>
+        <SearchLink
+          params={{ sex: 'f' }}
+          className={classNames({ 'is-active': sex === 'f' })}
+        >
+          Female
+        </SearchLink>
+      </p>
+
+      <div className="panel-block">
+        <p className="control has-icons-left">
+          <input
+            data-cy="NameFilter"
+            type="search"
+            className="input"
+            placeholder="Search"
+            value={query}
+            onChange={onQueryChange}
+          />
+
+          <span className="icon is-left">
+            <i className="fas fa-search" aria-hidden="true" />
+          </span>
+        </p>
+      </div>
+
+      <div className="panel-block">
+        <div className="level is-flex-grow-1 is-mobile" data-cy="CenturyFilter">
+          <div className="level-left">
+            {centuriesList.map(century => (
+              <SearchLink
+                key={century}
+                params={{
+                  centuries: (
+                    centuries.includes(century.toString())
+                      ? centuries.filter(time => time !== century.toString())
+                      : [...centuries, century]
```

Comment:
Move to variable long expression. It will be in 67 line. If you have any questions, please ask them in fe_chat and tag mentors. It will be faster, because your answers in GitHub will be visible to mentor only after you re-request review